spec/authorization_spec.rb

```diff
@@ -59,4 +59,80 @@
       end.should raise_exception(RestClient::Unauthorized)
     end
 
+  it "super admin endpoints should reject activation key auth" do
+    no_auth_client = Candlepin.new
+    @cp.create_activation_key(@owner['key'], 'key1')
+
+    lambda do
+      no_auth_client.create_owner_with_activation_key(random_string("test_owner"), @owner['key'], ['key1'])
+    end.should raise_exception(RestClient::Forbidden)
+  end
+
+  it "verified endpoints should reject activation key auth" do
+    no_auth_client = Candlepin.new
+    @cp.create_activation_key(@owner['key'], 'key1')
+
+    lambda do
+      no_auth_client.get_owner_with_activation_key(@owner['key'], ['key1'])
+    end.should raise_exception(RestClient::Forbidden)
+  end
+
+  it "default security hole should reject activation key auth" do
+    no_auth_client = Candlepin.new
+    @cp.create_activation_key(@owner['key'], 'key1')
+
+    lambda do
+      owner = no_auth_client.get_product_with_activation_key(@owner['key'], 'test_prod', ['key1'])
+    end.should raise_exception(RestClient::Forbidden)
+  end
+
+  it "activation key auth requires owner" do
+    no_auth_client = Candlepin.new
+
+    lambda do
+      no_auth_client.create_owner_with_activation_key(random_string("test_owner"), nil, ['key1'])
```

Comment:
This test, as well as the 4 next ones, should be testing against the registration endpoint instead of `create_owner_with_activation_key` to be more accurate with real world use. The previous 3 tests have already covered authorization against other types of endpoints

Reply:
Done.